### PR TITLE
LPD-28139 - React 18 Upgrade

### PR DESCRIPTION
**Pull request by bryceosterhaus**
Creating another PR to test CI in parallel

### CODE REVIEW

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:report:19955864

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:report:21572930

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:test:workflow

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:report:24269490

**Comment by bryceosterhaus** [comment]
ci:report:24269476

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:report:24773116

**Comment by bryceosterhaus** [comment]
ci:test:workflow

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/utils/types.ts [hunk 1]
```diff
@@ -38,6 +38,9 @@ export interface IBaseVisualizationMode<Mode extends string> {
 	thumbnail: string;
 	visualizationModeId: string;
 }
+export interface ICards extends IBaseVisualizationMode<'cards'> {}
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/utils/types.ts:41]: Duplicated interface export

**Comment by bryceosterhaus** [comment]
ci:report:26558342

**Comment by bryceosterhaus** [comment]
ci:test:relevant

File: workspaces/liferay-testray-workspace/client-extensions/liferay-testray-custom-element/src/context/TestrayContext.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
  */
 
-import {ReactNode, createContext, useEffect, useMemo, useReducer} from 'react';
+import {createContext, useEffect, useMemo, useReducer} from 'react';
```
Comment (juanjofgliferay) [inline workspaces/liferay-testray-workspace/client-extensions/liferay-testray-custom-element/src/context/TestrayContext.tsx:8]: Should we add `React.ReacNode` in [line 120](https://github.com/liferay-frontend/liferay-portal/pull/4441/files#diff-a388ae1326c3bd61a1c27a2cc260ef3dee5ee514bbdd7c8dd8193df790ef0dfaR120)?

File: modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/render.tsx [hunk 3]
```diff
@@ -88,7 +122,12 @@ export default function render(
 					 * can be found.
 					 */
 					try {
-						ReactDOM.unmountComponentAtNode(container);
+						if (USE_REACT_16) {
+							ReactDOM.unmountComponentAtNode(container);
```
Comment (juanjofgliferay) [inline modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/render.tsx:126]: Got warnings related to using `ReactDOM.unmountComponentAtNode` with version 18. Found some occurrences in the portal. Does this mean that the owner of the component should take care of it?

Screenshot from 2024-09-30 13-14-07

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx [hunk 4]
```diff
@@ -452,6 +458,9 @@ const OrderableTable = ({
 				</ManagementToolbar.Container>
 
 				{items.length ? (
+
+					// @ts-ignore
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx:462]: Anything we should consider here? Removing the comment does not throw any warning or error

Comment (bryceosterhaus) [reply modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx:462]: If the comment is removed, you get this type error

Screenshot 2024-10-01 at 9 56 55 AM

**Comment by bryceosterhaus** [comment]
ci:report:26981757

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:test:forms

**Comment by bryceosterhaus** [comment]
ci:report:30571559

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:report:31338664

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:report:32071533

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:report:33885204

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:report:44558391

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:report:48792161

**Comment by bryceosterhaus** [comment]
resent here, https://github.com/liferay-frontend/liferay-portal/pull/4507

**Pull request closed without merging**